xbmc/interfaces/legacy/AddonUtils.h

```diff
@@ -68,6 +69,9 @@ namespace XBMCAddonUtils
   };
 
 #define LOCKGUI XBMCAddonUtils::GuiLock __gl
+#define LOCKGUIIF(cond) std::unique_ptr<XBMCAddonUtils::GuiLock> __gl; \
```

Comment:
LOCKGUI unlocks GIL as we do in every other python interface funtion. Now with offscreen use, GIL is not unlocked. Calling into Kodi without unlocking GIL can lead to deadlocks.